app/scripts/models/user.js

```diff
@@ -532,12 +532,25 @@ define(function (require, exports, module) {
     },
 
     /**
+     * Fetch the clients for the given account, populated the passed in
+     * Clients collection.
+     *
+     * @param {Object} account - account for which device list is requested
+     * @param {Object} apps - Clients collection used to store list.
+     * @returns {Promise} resolves when the action completes
+     */
+    fetchAccountApps: function (account, apps) {
+      return this._oAuthClient.fetchClients(account.get('accessToken'), apps)
```

Comment:
Passing in `apps` seems a bit backwards, it seems like control should be inverted where the apps collection calls `user.fetchAccountApps` or `account.fetchApps` and then it takes responsibility for updating itself with the returned data.